test/helper.go

```diff
@@ -62,6 +62,17 @@ func ConnectSimple(ctx context.Context, t *testing.T, connString string) *pgx.Co
 	return conn.Conn()
 }
 
+func ConnectReplication(ctx context.Context, t *testing.T, connString string) *pgconn.PgConn {
+	is := is.New(t)
+
+	conn, err := pgconn.Connect(ctx, connString+"&replication=database")
+	is.NoErr(err)
+	t.Cleanup(func() {
+		is.NoErr(conn.Close(ctx))
```

Comment:
The test context might get cancelled, we still want to execute `Close` using an open context in that case.
```suggestion
		is.NoErr(conn.Close(context.Background()))
```